.github/workflows/test.yml

```diff
@@ -48,60 +42,20 @@ jobs:
         run: |
           source ./scripts/devenv.sh
           cd light-system-programs
-          yarn test-${{ matrix.test }}
-
-  light-zk-js:
-    name: light-zk.js
-    runs-on: buildjet-16vcpu-ubuntu-2204
-    steps:
-      - name: Checkout sources
-        uses: actions/checkout@v2
-
-      - name: Setup and build
-        uses: ./.github/actions/setup-and-build
-
-      - name: Test
-        run: |
-          source ./scripts/devenv.sh
-          cd light-zk.js
-          yarn test
-
-
-  mock-app-verifier:
-    name: mock-app-verifier
-    runs-on: buildjet-16vcpu-ubuntu-2204
-    steps:
-      - name: Checkout sources
-        uses: actions/checkout@v2
-
-      - name: Setup and build
-        uses: ./.github/actions/setup-and-build
-
-      - name: Test
-        run: |
-          source ./scripts/devenv.sh
-          cd mock-app-verifier
-          yarn test
-
-  light-circuits:
-    name: light-circuits
-    runs-on: buildjet-16vcpu-ubuntu-2204
-    steps:
-      - name: Checkout sources
-        uses: actions/checkout@v2
-
-      - name: Setup and build
-        uses: ./.github/actions/setup-and-build
-
-      - name: Test
-        run: |
-          source ./scripts/devenv.sh
-          cd light-circuits
-          yarn run test
-
-  relayer:
-    name: relayer
+          IFS=', ' read -r -a sub_tests <<< "${{ join(fromJSON(matrix['sub-tests']), ', ') }}"
+          for subtest in "${sub_tests[@]}"
+          do
+            yarn test-$subtest
```

Comment:
What's the goal here? We are merging multiple subtests into one job, because of setup taking too long?

Reply:
yes, because setup takes 5-8min while most tests take a lot less than that.
Also we can run up to 4 actions in parallel. Ideally this takes ci test time down to 20min.